R/zchunk_L2072.ag_water_irr_mgmt.R

```diff
@@ -1,67 +1,322 @@
+             minicam.energy.input = set_water_input_name(water_sector, water_type, A03.sector, GLU = GLU_name)) %>%
+      # Remove extra columns
+      select(one_of(LEVEL2_DATA_NAMES[["AgCoef"]])) ->
+      L2072.AgCoef_IrrWaterCons_bio_mgmt
+
+    # Create tables for water withdrawals IO coefs (consumption divided by irrigation efficiency)
+    L165.ag_IrrEff_R %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      right_join(bind_rows(L2072.AgCoef_IrrWaterCons_ag_mgmt, L2072.AgCoef_IrrWaterCons_bio_mgmt), by = "region") %>%
+      # Calculate water withdrawals using consumption divided by irrigation efficiency
+      mutate(coefficient = round(coefficient / field.eff, aglu.DIGITS_CALOUTPUT),
+             # Replace the last character "C" with "W" for input name
+             minicam.energy.input = gsub(".{1}$", "W", minicam.energy.input)) %>%
+      # Remove extra columns
+      select(one_of(LEVEL2_DATA_NAMES[["AgCoef"]])) ->
+      L2072.AgCoef_IrrWaterWdraw_mgmt
+
+    # Separate table of water withdrawals IO coefs for dedicated bioenergy crops
+    L2072.AgCoef_IrrWaterWdraw_mgmt %>%
+      filter(AgSupplySector == "biomass") ->
+      L2072.AgCoef_IrrWaterWdraw_bio_mgmt
+
+    # Ad hoc adjustment to water coefficients so that none of the region/GLU/crops have negative profit
+    # In GCAM, the profit rate is calculated as price minus cost times yield, so if the cost exceeds the price, then the profit goes negative.
+    # By adding in the water cost without modifying the non-land variable costs, we risk having costs that exceed commodity prices, which will
+    # cause solution/calibration failure in base years, and zero share in future years. This calculation checks whether any of our costs exceed
+    # the prices, and reduces water withdrawal coefficients where necessary.
+    # Note that the default unlimited water price is divided by the conveyance efficiency in order to replicate the prices in the model
+    L165.ag_IrrEff_R %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      # Separate table of water withdrawals IO coefs for primary crops
+      right_join(filter(L2072.AgCoef_IrrWaterWdraw_mgmt, AgSupplySector != "biomass"), by = "region") %>%
+      # Calculate water price and water cost
+      mutate(WaterPrice = DEFAULT_UNLIMITED_IRR_WATER_PRICE / conveyance.eff,
+             WaterCost = coefficient * WaterPrice) %>%
+      # Match in non-land variable costs
+      left_join_error_no_match(L2052.AgCost_ag_irr_mgmt, by = c("region", "AgSupplySector", "AgSupplySubsector", "AgProductionTechnology", "year")) %>%
+      # Match in commodity price
+      left_join_error_no_match(L132.ag_an_For_Prices, by = c("AgSupplySector" = "GCAM_commodity")) %>%
+      # Calculate profit as commodity price minus water costs and non-land variable costs
+      mutate(Profit = calPrice - WaterCost - nonLandVariableCost) ->
+      L2072.AgCoef_IrrWaterWdraw_ag_mgmt
```

Comment:
This seems like the wrong name for this tibble. Wouldn't `Profit` be better than `Coef`?